XenAdmin/Wizards/NewSRWizard_Pages/ChooseSrProvisioningPage.cs

```diff
@@ -67,26 +69,58 @@ public bool IsGfs2
 
         public override void PopulatePage()
         {
+            Connection.Cache.RegisterCollectionChanged<Cluster>(Cluster_CollectionChangedWithInvoke);

```

Comment:
Thinking more about this, we should only register the CollectionChanged event on PageLoaded and deregister it on PageLeave. This way the page won't be refreshed when it's not visible. 
We will also need to reload the page on PageLoaded, not on PagePopulate (which is only called on the fisrt load).
Plus, on each reload we shouldn't change the selection of the radio buttons (currently we select first radio button even on reload).